src/pages/package-editor.tsx

```diff
@@ -17,6 +19,16 @@ export const EditorPage = () => {
   const uuid4RegExp = new RegExp(
     /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$/,
   )
+  const session = useGlobalStore((s) => s.session)
+
+  if (
+    !isLoading &&
+    pkg?.is_private &&
+    pkg?.owner_github_username !== session?.github_username
+  ) {
+    return <NotFoundPage heading="Package Not Found" />
+  }
```

Comment:
Still I am not sure why the frontend know who the owner of that package is or the package is private or not. This doesn't need to happen here